File: principles-of-programming-languages/assignment-1/assignment-1/PrinciplesOfProgrammingLanguagesAssignment1/src/PrinciplesOfProgrammingLanguagesAssignment1/Kisi.java
// @ Rümeysa Ceran 

package PrinciplesOfProgrammingLanguagesAssignment1;

public class Kisi {
	String ad_soyad;
	int yas;
	static int ID = 0;
	int id;

	public Kisi(String ad_soyad, int yas) {
		this.ad_soyad = ad_soyad;
		this.yas = yas;
		this.id = ID;
		ID++;
	}
}

File: principles-of-programming-languages/assignment-2/assignment-2/PrinciplesOfProgrammingLanguagesAssignment2/src/Kisi.c
// @ Rümeysa Ceran 

#include "Kisi.h"
#include <stdlib.h>

static int ID = 0;

Kisi KisiOlusturma(char* ad_soyad, int yas){
    Kisi this;
    this = (Kisi)malloc(sizeof(struct KISI));
    this->ad_soyad = ad_soyad;
    this->yas = yas;
    this->yoket = &KisiYoket;
    this->id = ++ID;
    return this;
}

void KisiYoket(const Kisi this){
    if(this == NULL) return;
    free(this->ad_soyad);
    free(this);
}

//int KisiSayisi(int a){
//    return ++a;
//}